src/main/app/documents/documentsClient.ts

```diff
@@ -1,24 +1,72 @@
 import * as config from 'config'
-import { requestNonPromise } from 'app/client/request'
+import { request, requestNonPromise } from 'client/request'
 import * as http from 'http'
 import StringUtils from 'app/utils/stringUtils'
+import * as fs from 'fs'
+import * as URL from 'url-parse'
 
 const claimStoreBaseUrl = config.get<string>('claim-store.url')
+const documentManagementUrl = config.get<string>('documentManagement.url')
 
 export default class DocumentsClient {
 
-  constructor (public documentsUrl: string = `${claimStoreBaseUrl}/documents`) {
-    this.documentsUrl = documentsUrl
-  }
-
-  getSealedClaim (claimExternalId: string, bearerToken: string): http.IncomingMessage {
+  static getSealedClaim (claimExternalId: string, userAuthToken: string): http.IncomingMessage {
     if (StringUtils.isBlank(claimExternalId)) {
       throw new Error('Claim external ID cannot be blank')
     }
     return requestNonPromise.get({
-      uri: `${this.documentsUrl}/legalSealedClaim/${claimExternalId}`,
+      uri: `${claimStoreBaseUrl}/documents/legalSealedClaim/${claimExternalId}`,
+      headers: {
+        Authorization: `Bearer ${userAuthToken}`
+      }
+    })
+  }
+
+  static save (userAuthToken: string, file: any): Promise<string> {
+
+    const endpointURL: string = `${documentManagementUrl}/documents`
+
+    return request.post(endpointURL, {
+      headers: {
+        Authorization: `Bearer ${userAuthToken}`
+      },
+      formData: {
+        files: [
+          {
+            value: fs.createReadStream(file.path),
+            options: { filename: file.name, contentType: file.type }
+          }
+        ],
+        classification: 'PRIVATE'
+      }
+    })
+      .then((response: any) => {
+        const href: string = response._embedded.documents[0]._links.self.href
+        return new URL(href).pathname
+      })
+  }
+
+  static getBinaryUrl (userAuthToken: string, path: string): Promise<string> {
+
+    const endpointURL: string = `${documentManagementUrl}${path}`
```

Comment:
Have you checked how the endpoint will handle double `/` in case that happens sometime? E.g. `http://document-mgmt-host//path-to-binary`. Looks like something that can break accidentally, might be worth to include a check/URL sanitising function here.

Reply:
should maybe be part of the unit tests for documentsClient to make sure that it fails if the path contains a / ? The only places that path get set should be from in our code

Reply:
unnecessary imo, unit tests would add no value as:
1. you rely on the contract that `URL(href).pathname` parses correctly, safe assumption
2. the URL you are given comes from a trusted API which uses HATEOAS which relies on passing URLs around, the only reason for doing  URL + PATH, instead of taking the URL given to us by them is <docker> vs <localhost> issues.